embassy-stm32/src/usart/mod.rs

```diff
@@ -106,8 +106,8 @@ impl<'d, T: Instance, TxDma> UartTx<'d, T, TxDma> {
                 reg.set_dmat(true);
             });
         }
-        let dst = tdr(T::regs());
-        crate::dma::write(ch, request, buffer, dst).await;
+        let transfer = crate::dma::write(ch, request, buffer, tdr(T::regs()));
```

Comment:
why the temp variable? I think `write(...).await;` should work

Reply:
wat? :D

Reply:
I mean, maybe the `write(ch, request, buffer, tdr(T::regs()))` is internally expanded by compiler as:
```rust
let _tmp1 = T::regs();
let _tmp2 = tdr(_tmp1);
write(ch, request, buffer, _tmp2);
```
I'm not a compiler guru so IDK :D